src/core/execute.c

```diff
@@ -4822,6 +4822,16 @@ int exec_context_get_clean_directories(
                         r = strv_consume(&l, j);
                         if (r < 0)
                                 return r;
+
+                        if (exec_directory_is_private(c, t)) {
+                                j = path_join(prefix[t], "private", *i);
+                                if (!j)
+                                        return -ENOMEM;
+
+                                r = strv_consume(&l, j);
+                                if (r < 0)
+                                        return r;
+                        }
```

Comment:
hmm, an alternative might be two unconditionally delete both the regular and the private dir. Sounds like it would be more robust during updates?

Reply:
OK. I agree with you.